Java/core/src/main/java/com/amazon/randomcutforest/RandomCutForest.java

```diff
@@ -202,19 +229,38 @@
     public RandomCutForest(Builder<?> builder) {
         this(builder, false);
         rng = builder.getRandom();
-        if (compactEnabled) {
+        if (floatEnabled) {
+            initCompactFloat();
+        } else if (compactEnabled) {
             initCompact();
         } else {
             initNoncompact();
         }
     }
 
     private void initCompact() {
```

Comment:
Should we change this to initCompactDouble() for consistency?

Reply:
Done.